components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -629,6 +629,20 @@ private boolean validateScope(OAuthTokenReqMessageContext tokReqMsgCtx) throws I
 
         OAuth2AccessTokenReqDTO tokenReqDTO = tokReqMsgCtx.getOauth2AccessTokenReqDTO();
         String grantType = tokenReqDTO.getGrantType();
+        if (tokReqMsgCtx.isImpersonationRequest() && OAuthConstants.GrantTypes.TOKEN_EXCHANGE.equals(grantType)) {
+            /*
+             In the impersonation flow, we have already completed scope validation during the /authorize call and
```

Comment:
What if we filter out the intersection of the scopes that passed through the subject token and the token request?

Reply:
If the scope body param present in the token exchange request, we should get the intersection. This will be handled in https://github.com/wso2-extensions/identity-oauth2-grant-token-exchange/pull/26/files#diff-2d5991ab029cd499dc04f0cb2adeb35f782d03473b7c86d9392cc7e98d78ba9bR241